codec-redis/src/main/java/io/netty/handler/codec/redis/RedisMessage.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License, version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.netty.handler.codec.redis;
+
+/**
+ * RedisMessage is base interface for codec-redis.
+ */
+public interface RedisMessage {
+
+    /**
+     * Returns the {@link RedisMessageType}.
+     *
+     * @return the message type.
+     */
+    RedisMessageType type();
```

Comment:
@jongyeol - Thanks for the background. Yes I think we should go one way or the other. It seems right now the encoder/decoder (and a few other places) are dependent on the internals of the different types. Other Netty codecs also use the type hierarchy in a similar fashion, so maybe we should commit to the type hierarchy, unless there is a clear benefit to having `type()` here.

@mp911de - WDYT?